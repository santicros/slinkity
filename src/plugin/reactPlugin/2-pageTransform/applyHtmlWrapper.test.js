const { parse } = require('node-html-parser')
const applyHtmlWrapper = require('./applyHtmlWrapper')

describe('applyHtmlWrapper', () => {
  it('should apply missing <html>,<head>, and <body> to fragment', () => {
    const root = parse('<ul id="list"><li>Hello World</li></ul>')
    applyHtmlWrapper(root)
    expect(root.outerHTML).toMatchSnapshot()
  })
  it('should not apply <body> when <body> is already present', () => {
    const root = parse('<body><p>Content!</p></body>')
    applyHtmlWrapper(root)
    expect(root.outerHTML).toMatchSnapshot()
  })
  it('should not apply <head> when <head> is already present', () => {
    const root = parse(`<head>
  <title>I'm already here!</title>
</head>
<body>
  <ul id="list"><li>Hello World</li></ul>
</body>`)
    applyHtmlWrapper(root)
    expect(root.outerHTML).toMatchSnapshot()
  })
  it('should not do anything when <html> is already present', () => {
    const root = parse(`<html>
<head>
  <title>I'm already here!</title>
</head>
<body>
  <ul id="list"><li>Hello World</li></ul>
</body>
</html>`)

    applyHtmlWrapper(root)
    expect(root.outerHTML).toMatchSnapshot()
  })
})
