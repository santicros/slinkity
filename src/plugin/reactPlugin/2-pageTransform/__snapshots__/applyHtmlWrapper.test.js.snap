// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyHtmlWrapper should apply missing <html>,<head>, and <body> to fragment 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <title>Document</title>
</head>

<body>
  <ul id=\\"list\\"><li>Hello World</li></ul>
</body>

</html>"
`;

exports[`applyHtmlWrapper should not apply <body> when <body> is already present 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <title>Document</title>
</head>

<body><p>Content!</p></body>

</html>"
`;

exports[`applyHtmlWrapper should not apply <head> when <head> is already present 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

<head>
  <title>I'm already here!</title>
</head>
<body>
  <ul id=\\"list\\"><li>Hello World</li></ul>
</body>

</html>"
`;

exports[`applyHtmlWrapper should not do anything when <html> is already present 1`] = `
"<html>
<head>
  <title>I'm already here!</title>
</head>
<body>
  <ul id=\\"list\\"><li>Hello World</li></ul>
</body>
</html>"
`;
